### DISCUSSION TITLE
Clang's Static Analyser Warnings

### DISCUSSION THREAD

**Discussion by psi29a** [bulletphysics/bullet3#3467] [General]
I'm sure there are more, but here the ones that really caught my eye while compiling OpenMW:

<pre>
/usr/include/bullet/LinearMath/btVector3.h:1140:4: warning: Value stored to 'maxVal' is never read
                        maxVal = m_floats[3];
                        ^        ~~~~~~~~~~~
/usr/include/bullet/LinearMath/btVector3.h:1169:4: warning: Value stored to 'minVal' is never read
                        minVal = m_floats[3];
                        ^        ~~~~~~~~~~~
In file included from /media/workspace/OpenMW/openmw/apps/openmw/mwclass/weapon.cpp:15:
In file included from /media/workspace/OpenMW/openmw/apps/openmw/mwclass/../mwworld/physicssystem.hpp:6:
In file included from /usr/include/bullet/btBulletCollisionCommon.h:35:
In file included from /usr/include/bullet/BulletCollision/CollisionShapes/btBvhTriangleMeshShape.h:22:
In file included from /usr/include/bullet/BulletCollision/CollisionShapes/btTriangleInfoMap.h:21:
/usr/include/bullet/LinearMath/btSerializer.h:314:4: warning: Value stored to 'cp' is never read
                        cp = (char*)intPtr;
                        ^    ~~~~~~~~~~~~~
3 warnings generated.
</pre>

This was with 2.81, but from what I see, it is still here in master.

**Comment by psi29a** [comment]
Here is the full result (5.7MiB bz2) of the Clang's Static Analyzer run on the master branch at commit 46bd05f4f769ae0c6229489bb515ea6e471f17b7 (Sep 17):

https://dl.dropboxusercontent.com/u/396161/scan-build-2014-10-01-094545-6237-1.tar.bz2

It contains a lot of low-hanging fruit.

**Comment by erwincoumans** [comment]
Thanks for the report, I hope to find some time to check it out and fix issues/warnings

**Comment by psi29a** [comment]
I've begone working through the list, first just the dead code areas:
https://github.com/bulletphysics/bullet3/pull/256
